db/migrate/20150713143226_add_unique_to_operatingsystems_title.rb

```diff
@@ -0,0 +1,17 @@
+class AddUniqueToOperatingsystemsTitle < ActiveRecord::Migration
+  def up
+    duplicities = Operatingsystem.all.group_by(&:title).values.keep_if { |ary| ary.length > 1}
+    duplicities.each do |systems|
+      systems.each do |os|
+        os.description = os.fullname
```

Comment:
You are right, fullname is not enforced at DB level, though all records inserted through Foreman (and not changed directly) are ok.

At the moment, I can't think of anything else than direct modification of DB that could cause this.